src/blocks/battery.rs

```diff
@@ -215,6 +229,30 @@ impl Block for Battery {
         vec![&self.output]
     }
 
+    fn click(&mut self, event: &I3BarEvent) -> Result<()> {
```

Comment:
Seems like all of these functions are the same, so it mgiht be a better idea to move it out from impl (maybe to ButtonWidget?) with a signature like `fn handle_click(id: &String, on_click: &String, event: &I3BarEvent) -> Result<()>` and for the impl left just a call for `handle_click(&self.id, &self.on_click, event)`